test/chrome/crdpMultiplexing/crdpMultiplexor.test.ts

```diff
@@ -0,0 +1,175 @@
+    removeListener(event: string, cb: Function): any { /* empty */ }
+    on(event: 'open', cb: (ws: LikeSocket) => void): any;
+    on(event: 'message', cb: (data: string) => void): any;
+    on(event: string, cb: Function): any { /* empty */ }
+}
+
+suite('CRDPMultiplexor', () => {
+
+    let multiplexor: CRDPMultiplexor;
+    let webSocketMock: Mock<StubSocket>;
+    let socketMessageCallbacks: Function[];
+
+
+    setup(() => {
+        let socket = new StubSocket();
+        socketMessageCallbacks = [];
+        webSocketMock = Mock.ofInstance<StubSocket>(socket);
+        webSocketMock
+            .setup(s => s.on('message', It.isAny()))
+            .returns((s: string, cb: any) => {
+                socketMessageCallbacks.push(cb);
+            });
+        webSocketMock
+            .setup(s => s.send(It.isAnyString()))
+            .returns((s: string) => {
+                let message = JSON.parse(s);
+                for (let cb of socketMessageCallbacks) {
+                    // Send empty result once we received a message
+                    cb('{"id":' + message.id + ', "result":{}}');
+                }
+            });
+        multiplexor = new CRDPMultiplexor(webSocketMock.object);
+    });
+
+    teardown(() => {
+        multiplexor = undefined;
+        webSocketMock.reset();
+        webSocketMock = undefined;
+        socketMessageCallbacks = [];
+    });
+
+    test('Channel is successfully added', done => {
+        let testChannel = multiplexor.addChannel("testChannel");
+        Assert.notEqual(testChannel, null, "Test channel should not be null");
+        done();
+    });
+
+    test('Multiplexor message sent to underlying socket and response received.', done => {
+        let channel = multiplexor.addChannel("testChannel");
+        let messageId = 1;
+        let channelCallback = (data: string) => {
+            // Test receiving data - called when socket sends back response data
+            Assert.equal(JSON.parse(data).id, messageId);
+            done();
+        };
+
+        // Test sending data
+        channel.on("message", channelCallback);
+        multiplexor.send(channel, '{"method":"Runtime.enable","id":' + messageId + '}');
+        webSocketMock.verify(s => s.send(It.isAnyString()), Times.atLeastOnce());
+    });
+
+    test('Multiplexor message handles multiple channels', () => {
+        let callbackPromises: Promise<void>[] = [];
+
+        for (let i = 1; i < 5; i++) {
+            let channel = multiplexor.addChannel("Channel" + i);
+
+            let callbackPromise = new Promise<void>(function (resolve, reject) {
+                let callback = (data: string) => {
+                    Assert.equal(JSON.parse(data).id, i);
+                    resolve();
+                }
+
+                channel.on("message", callback);
+            });
+
+            callbackPromises.push(callbackPromise);
+            multiplexor.send(channel, '{"method":"Console.enable","id":' + i + '}');
```

Comment:
It would also be interesting to test that if all channels use the same message id, things still work